tasks/kernel_matrix_testing/libvirt.py

```diff
@@ -1,11 +1,20 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable, Iterable, List
+
 from tasks.kernel_matrix_testing.tool import info
 
+if TYPE_CHECKING:
```

Comment:
Would this not always be true, since you are importing this above?

Reply:
`TYPE_CHECKING` is true only when the type validator is checking things. In runtime it's false, which means that the types are not imported. This is necessary because not all CI jobs have the required packages (`typing_extensions`) or Python version (3.10+) to import the `types.py` file. This allows us to have type information without having extra runtime requisites.